std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoSecretHelper.java

```diff
@@ -177,10 +179,43 @@ public EnsoHttpResponse reconstructResponseFromCachedStream(
   }
 
   public static EnsoHTTPResponseCache getOrCreateCache() {
-    if (cache == null) {
-      cache = new EnsoHTTPResponseCache();
+    if (getCache() instanceof EnsoHTTPResponseCache httpCache) {
+      return httpCache;
+    } else {
+      var module =
+          Context.getCurrent()
+              .eval(
+                  "enso",
+                  """
+      import Standard.Base.Runtime.Managed_Resource.Managed_Resource
+      import Standard.Base.Data.Boolean.Boolean
+
+      type Cache
+          private Value ref:Managed_Resource
+
+          new obj -> Cache =
+              on_finalize _ = 0
```

Comment:
@GregoryTravis - At the end I decided to use enhanced `Managed_Resource` as it already has support for GC and finalizers. Feel free to implement a cleanup logic in this `on_finalize` method. 

It things seem to work for you and I get approvals from engine guys, let's integrate!

Reply:
I have implemented it [here](https://github.com/enso-org/enso/pull/11673/files#diff-eb57474c1b9fa4337971c6919b17b13627f2e8e9200f35744b52bbc064673dfeR53). The test passes and it works in the GUI.

Reply:
I am still using a `Ref`, but now I am clearing it myself, using `on_finalize`, and using that to respond to the reload button.

Reply:
> still using a Ref, but now I am clearing it myself, using on_finalize

It is not a bad solution. The `on_finalize` may be called at _"any moment"_ (like interrupts do) and the less it does, the better.